adapter/internal/api/restserver/config_restapi.go

```diff
@@ -59,29 +61,61 @@ func configureAPI(api *operations.RestapiAPI) http.Handler {
 	api.JSONProducer = runtime.JSONProducer()
 
 	// Applies when the Authorization header is set with the Basic scheme
-	api.BasicAuthAuth = func(user string, pass string) (*models.Principal, error) {
-		authenticated := false
-		for _, regUser := range mgwConfig.Adapter.Server.Users {
-			if user == regUser.Username && pass == regUser.Password {
-				authenticated = true
-			}
-		}
-		if !authenticated {
-			logger.LoggerAPI.Info("Credentials provided for deploy command are invalid")
+	api.BasicAuthAuth = func(username, password string) (*models.Principal, error) {
+		validCredentials := auth.ValidateCredentials(username, password, mgwConfig)
+		if !validCredentials {
+			logger.LoggerAPI.Info("Credentials provided for basic auth are invalid")
 			return nil, errors.New(401, "Credentials are invalid")
 		}
 		// TODO: implement authentication principal
 		p := models.Principal{
 			Token:    "xxxx",
 			Tenant:   "xxxx",
-			Username: user,
+			Username: username,
 		}
 		logger.LoggerAPI.Debugf("Principal : %v", p)
 		return &p, nil
 	}
 
-	api.APIIndividualDeleteApisHandler = api_individual.DeleteApisHandlerFunc(func(params api_individual.DeleteApisParams,
-		principal *models.Principal) middleware.Responder {
+	api.BearerTokenAuth = func(token string, scopes []string) (*models.Principal, error) {
+		valid, err := auth.ValidateToken(token, scopes)
+		if err != nil || !valid {
+			logger.LoggerAPI.Info("The provided token is not valid or server error")
+			return nil, errors.Unauthenticated("The provided token is not valid or server error")
+		}
+		p := models.Principal{
+			Token:    token,
+			Tenant:   "xxxx",
```

Comment:
what does `xxxx` mean. can't this be nil or empty?

Reply:
I am not exactly sure. It was there for basic auth, therefore I used the same with the token

Reply:
remove tenant